File: examples/Delphi/ExaBtn1.pas
unit Exabtn1;

interface

uses
  SysUtils, WinTypes, WinProcs, Messages, Classes, Graphics, Controls,
  Forms, Dialogs, StdCtrls, Buttons, OvcABtn, OvcBase, OvcEF, OvcPB, OvcPF,
  OvcDate;

type
  TForm1 = class(TForm)
    BitBtn1: TBitBtn;
    ABtn: TOvcAttachedButton;
    PF: TOvcPictureField;
    OvcController1: TOvcController;
    procedure ABtnClick(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  Form1: TForm1;

implementation

uses
  ExABtn2;

{$R *.DFM}

procedure TForm1.ABtnClick(Sender: TObject);
var
  P : TPoint;
begin
  P := ClientToScreen(Point(ABtn.Left+ABtn.Width , ABtn.Top + ABtn.Height));
  frmCal.Left := P.X - frmCal.Width;
  frmCal.Top := P.Y;
  frmCal.Calendar.CalendarDate := PF.AsDateTime;
  if frmCal.ShowModal = mrOK then
    PF.AsStDate := DateTimeToSTDate(frmCal.Calendar.CalendarDate);
end;

end.

File: examples/CBuildr6/ExABtn1.cpp
//---------------------------------------------------------------------------
#include <vcl\vcl.h>
#pragma hdrstop

#include "ExABtn1.h"
#include "ExABtn2.h"
//---------------------------------------------------------------------------
#pragma link "OvcABtn"
#pragma link "OvcPF"
#pragma link "OvcPB"
#pragma link "OvcEF"
#pragma link "OvcBase"
#pragma resource "*.dfm"
TForm1 *Form1;
//---------------------------------------------------------------------------
__fastcall TForm1::TForm1(TComponent* Owner)
  : TForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TForm1::ABtnClick(TObject *Sender)
{
  TPoint P;

  P = ClientToScreen(Point(ABtn->Left+ABtn->Width , ABtn->Top + ABtn->Height));
  frmCal->Left = P.x - frmCal->Width;
  frmCal->Top = P.y;
  frmCal->Calendar->CalendarDate = PF->AsStDate;
  if (frmCal->ShowModal() == mrOk)
    PF->AsStDate = frmCal->Calendar->CalendarDate;

}
//---------------------------------------------------------------------------
